### PR TITLE
CP-38529: Replace built-in PuTTY with user-provided location

### PR DESCRIPTION
**Pull request by danilo-delbusso**
Some automatic whitespace changes so I suggest you remove those diffs.

Tried to make it easy to review commit-by-commit.

You won't be able to see the "warning" version of the error dialog unless CHC can't find either client on first start-up. So best approach is probably to temporarily rename the `.exe`s.

Defaults paths are:

`C:\Windows\System32\OpenSSH\ssh.exe`
`C:\Program Files\PuTTY\putty.exe`

### CODE REVIEW

**Review by AaronRobson — APPROVED** [review]

File: XenModel/Messages.resx [hunk 1]
```diff
@@ -14737,4 +14737,28 @@ Any disk in your VM's DVD drive will be ejected when installing {1}.</value>
   <data name="YOU_ARE_HERE" xml:space="preserve">
     <value>You are here</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="EXTERNAL_TOOLS" xml:space="preserve">
```
Comment (MihaelaStoica) [inline XenModel/Messages.resx:14740]: We should try and keep Messages.resx sorted.

**Review by MihaelaStoica — APPROVED** [review]

**Comment by danilo-delbusso** [comment]
FYI: New commit isn't supposed to remove all docker-related code in `GeneralTabPage`, just the one that mentions PuTTY. Best reviewed without whitespace differences.

**Review by CitrixChris — APPROVED** [review]

**Review by AaronRobson — APPROVED** [review]

**Comment by danilo-delbusso** [comment]
Since this is a new feature, I'm awaiting @kc284's review. Will be flagging as `under review` for that reason. Don't merge please.

**Review by kc284 — DISMISSED** [review]
I've made some micro-corrections here https://github.com/danilo-delbusso/xenadmin/pull/14

**Review by CitrixChris — APPROVED** [review]

**Review by AaronRobson — APPROVED** [review]

**Pull request merged**